src/types.ts

```diff
@@ -206,3 +209,20 @@ export type NoOpTransaction = {
 export type TransportLayerOptions = {
   httpBodyPayload?: string;
 }
+
+export type VercelRequest<BodyType = JsonApiDocument> = IncomingMessage & {
```

Comment:
I think `VercelRequest` and `VercelResponse` types should be replaced with Next.js ones
```
import type { NextApiRequest, NextApiResponse } from 'next'

export default (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ name: 'John Doe' })
}
```

Reply:
Interesting, I hadn't noticed they're actually. I'll probably leave this one for Vercel (and not make any mentions to Next.js) and create a `jsonApiNext` middleware instead, to avoid confusion.

Reply:
I get it now, this is just for Vercel functions and not for Next. Ok, makes sense…

Reply:
At first I thought this could be used for both, but it's better not get them mixed up.